libqaul/src/api/mod.rs

```diff
@@ -126,13 +126,29 @@ impl Qaul {
     }
 
     /// Find a subset of contacts with some query
-    pub fn contacts_find(&self, user: UserAuth, query: String) -> QaulResult<Vec<User>> {
-        unimplemented!()
+    pub fn contacts_find<S: Into<String>>(
+        &self,
+        _user: UserAuth,
+        query: S,
+    ) -> QaulResult<Vec<User>> {
+        let users = self.users.lock().unwrap();
+        let query = query.into();
+        Ok(users
+            .iter()
+            .filter(|(_, user)| user.data.like_query(&query))
```

Comment:
Same thing here. Actually what we need first is a contact book store concept inside the user, before these endpoints can be implemented